Sources/FoundationInternationalization/Formatting/Number/BinaryInteger+FormatStyle.swift

```diff
@@ -43,250 +43,183 @@ extension BinaryInteger {
-                    // Note that for a word of zero (no remainder) this does technically overflow but it's intentional - zero is special since there's no distinct representation for -0 vs +0, but ~UInt(0) &+ 1 is 0, conveniently!  So we can trade a conditional (for avoiding this block if `word` is .zero) for two trivial arithmetic instructions.
-                    word = ~word &+ 1
-                }
-
-                let digitsAdded = word.numericStringRepresentation(intoEndOfBuffer: &buffer[...wordInsertionPoint])
-
-                if .zero != quotient { // Not on the last word, so need to fill in leading zeroes etc.
-                    wordInsertionPoint -= decimalDigitsPerWord
-
-                    let leadingZeroes = decimalDigitsPerWord - digitsAdded
-                    assert(0 <= leadingZeroes, "Negative leading zeroes \(leadingZeroes)!  Expected \(decimalDigitsPerWord) digits per word and added \(digitsAdded).")
-
-                    if 0 < leadingZeroes {
-                        buffer[(wordInsertionPoint + 1)...(wordInsertionPoint + leadingZeroes)].initialize(repeating: UInt8(ascii: "0"))
-                    }
-                } else {
-                    wordInsertionPoint -= digitsAdded
-                }
-
-                tmp = quotient
-            }
-
-            if negative {
-                buffer[wordInsertionPoint] = UInt8(ascii: "-")
-                wordInsertionPoint -= 1
-            }
-
-            actualDigits = wordInsertionPoint.distance(to: buffer.endIndex) - 1
-
-            let unusedDigits = maximumDigits - actualDigits
-            assert(0 <= unusedDigits, "Negative unused digits \(unusedDigits)!  Expected at most \(maximumDigits) digit(s) but emitted \(actualDigits).")
-
-            if 0 < unusedDigits {
-                buffer[0..<unusedDigits].initialize(repeating: 0) // The buffer is permitted to be partially uninitialised, but only at the end.  So we have to initialise the unused portion at the start, in principle.  Technically this probably doesn't matter given we never subsequently read this part of the buffer, but there's no way to express that such that the compiler can ensure it for us.
-            }
+        numericStringRepresentationForBinaryInteger(words: self.words, isSigned: Self.isSigned)
+    }
+}
 
-            initialisedCount = maximumDigits
-        }[(maximumDigits - actualDigits)...]
+/// Formats `words` in "Numeric string" format (https://speleotrove.com/decimal/daconvs.html)
+/// which is the required input form for certain ICU functions (e.g. `unum_formatDecimal`).
+///
+/// - Parameters:
+///   - words: The binary integer's words.
+///   - isSigned: The binary integer's signedness.
+///
+internal // In order to test big integers without big integer models.
```

Comment:
```suggestion
internal
```

Nit: I'd prefer leave out this kind of comment, especially when it's already an established pattern throughout the project